st2common/st2common/persistence/trigger.py

```diff
@@ -52,15 +52,24 @@ def delete_if_unreferenced(cls, model_object, publish=True, dispatch_trigger=Tru
         delete_query = model_object._object_key
         delete_query['ref_count__lte'] = 0
         cls._get_impl().delete_by_query(**delete_query)
+
+        # Since delete_by_query cannot tell if teh delete actually happened check with a get call
```

Comment:
That sucks.

Anyways, I was curious and looks like this was "improved" in newer versions of MongoDB (https://docs.mongodb.org/manual/reference/method/db.collection.remove/):

"Changed in version 2.6: The remove() returns an object that contains the status of the operation."

MongoDB 2.4 which we use is quite old so we should really consider upgrading in the near future...